apps/angular/projection/src/app/ui/card/card.component.ts

```diff
@@ -1,61 +1,34 @@
-import { NgFor, NgIf } from '@angular/common';
-import { Component, Input } from '@angular/core';
-import { randStudent, randTeacher } from '../../data-access/fake-http.service';
-import { StudentStore } from '../../data-access/student.store';
-import { TeacherStore } from '../../data-access/teacher.store';
-import { CardType } from '../../model/card.model';
+import { NgClass, NgFor, NgIf, NgTemplateOutlet } from '@angular/common';
+import { Component, EventEmitter, Input, Output, TemplateRef } from '@angular/core';
 import { ListItemComponent } from '../list-item/list-item.component';
 
 @Component({
   selector: 'app-card',
   template: `
     <div
-      class="flex w-fit flex-col gap-3 rounded-md border-2 border-black p-4"
-      [class]="customClass">
-      <img
-        *ngIf="type === CardType.TEACHER"
-        src="assets/img/teacher.png"
-        width="200px" />
-      <img
-        *ngIf="type === CardType.STUDENT"
-        src="assets/img/student.webp"
-        width="200px" />
-
+      class="flex w-fit flex-col gap-3 rounded-md border-2 border-black p-4" [class]="customClass">
+        <ng-content select="[cardImage]"></ng-content>
       <section>
-        <app-list-item
-          *ngFor="let item of list"
-          [name]="item.firstName"
-          [id]="item.id"
-          [type]="type"></app-list-item>
+        @for (item of list; track $index) {
```

Comment:
it's better to track on a unique property is possible.
if you change the list, the $index is not optimal since all index will change so the list will still get refresh entirely

Reply:
now I updated this line by `  @for (item of list; track item.id) {`, I hope now it track the changes in better way